test/python/unittest/API/APITest.py

```diff
@@ -0,0 +1,324 @@
+
+        chunk_data = E_x.load_chunk(offset, extent)
+        series.flush()
+
+        print("Chunk has been read from disk\n"
+              "Read chunk contains:")
+        print(chunk_data)
+
+    def testLoadSeries(self):
+        """ Test loading a pmd series from hdf5."""
+
+        # Get series.
+        series = self.__series
+        self.assertIsInstance(series, openPMD.Series)
+
+        print("Read a Series with openPMD standard version %s" %
+              series.openPMD)
+
+    def testIterations(self):
+        """ Test querying a series' iterations and loop over them. """
+
+        # Get series.
+        series = self.__series
+
+        # Loop over iterations.
+        self.assertIsInstance(series.iterations, openPMD.Iteration_Container)
+        self.assertTrue(hasattr(series.iterations, "__getitem__"))
+        for i in series.iterations:
+            # self.assertIsInstance(i, int)
+            self.assertIsInstance(series.iterations[i], openPMD.Iteration)
+
+        # Check type.
+        i = series.iterations[100]
+        self.assertIsInstance(i, openPMD.Iteration)
+
+    def testMeshes(self):
+        """ Test querying a mesh. """
+
+        # Get series.
+        series = self.__series
+
+        # Check type.
+        i = series.iterations[100]
+
+        # Check meshes are iterable.
+        self.assertTrue(hasattr(i.meshes, "__getitem__"))
+        for m in i.meshes:
+            # self.assertIsInstance(m, str)
+            self.assertIsInstance(i.meshes[m], openPMD.Mesh)
+
+    def testParticles(self):
+        """ Test querying a particle species. """
+
+        # Get series.
+        series = self.__series
+
+        # Check type.
+        i = series.iterations[100]
+
+        self.assertTrue(hasattr(i.particles, "__getitem__"))
+        for ps in i.particles:
+            # self.assertIsInstance(ps, str)
+            self.assertIsInstance(i.particles[ps], openPMD.ParticleSpecies)
+
+    def testData_Order(self):
+        """ Test openPMD.Data_Order. """
+        obj = openPMD.Data_Order('C')
+        del obj
+
+    def testDatatype(self):
+        """ Test openPMD.Datatype. """
+        data_type = openPMD.Datatype(1)
+        del data_type
+
+    def testDataset(self):
+        """ Test openPMD.Dataset. """
+        data_type = openPMD.Datatype(1)
+        extent = [1, 1, 1]
+        obj = openPMD.Dataset(data_type, extent)
+        del obj
+
+    def testGeometry(self):
+        """ Test openPMD.Geometry. """
+        obj = openPMD.Geometry(0)
+        del obj
+
+    def testIteration(self):
+        """ Test openPMD.Iteration. """
+        self.assertRaises(TypeError, openPMD.Iteration)
+
+        iteration = self.__particle_series.iterations[400]
+
+        copy_iteration = openPMD.Iteration(iteration)
+
+        self.assertIsInstance(copy_iteration, openPMD.Iteration)
+
+    def testIteration_Container(self):
+        """ Test openPMD.Iteration_Container. """
+        obj = openPMD.Iteration_Container()
```

Comment:
This is a mis-design in the C++ API. ```Container``` should not have a publicly available constructor. #280